test/specs/reference/tests/02.related_reference.js

```diff
@@ -470,6 +470,61 @@ exports.execute = function(options) {
                     var ref = pageWithToName.tuples[0].getAssociationRef({"not the id":9003});
                     expect(ref).toBe(null);
                 });
+
+                describe('Tuple.getBatchAssociationRef, ', function () {
+                    var batchPage, selectedTuples;
+                    it('should return an array of references including the filtered assocation reference.', function() {
+                        var url = options.url + "/catalog/" + catalog_id + "/entity/";
+                        // there are 3 rows in the set for "pageWithToName"
+                        related[2].sort([{"column":"id", "descending":false}]).read(3).then(function(response) {
+                            batchPage = response;
+                            expect(batchPage.tuples.length).toBe(3, "pageWithToName set is not of size 3");
+
+                            selectedTuples = [batchPage.tuples[0], batchPage.tuples[1]];
```

Comment:
Doesn't `Page.tuples` already return an array for you? Why do you need to do it this way?

Reply:
In this case, batchPage is length 3 but I'm only "selecting" 2 of them for the following test.